### ISSUE TITLE
[Bug]: Querynode encountered an OOMKilled error when load with enough memory

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#11004]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211101-881e547
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.0rc8.dev18
- OS(Ubuntu or CentOS): 
- CPU/Memory: limit querynode memory 2Gi
- GPU: 
- Others:
```

### Current Behavior

1. Deploy milvus cluster and limit querynode memory 2Gi
2. Load collection into querynode: `dim=512, num_entities=307200`, about 608Mi, one of the segment size is 303Mi
3. Load hangs and query node down because of OOMKilled
4. After querynode pod re-running, load still hangs.

### Expected Behavior

totalMem = 2048Mi
usedMem=50Mi
segmentTotalSize=608Mi
currentSegmentSize=303Mi
totalMem < usedMem + segmentTotalSize + currentSegmentSize*2, and 2048Mi < 1264Mi
I expected to load successfully

### Steps To Reproduce

```markdown
You can refer to the `test_chaos_memory_stress.py::TestChaosData::test_chaos_memory_stress_querynode` for the code
```

### Anything else?

The first querynode pod log: 
[querynode.log](https://github.com/milvus-io/milvus/files/7451959/querynode.log)

The second querynode pod log:
[querynode_1.log](https://github.com/milvus-io/milvus/files/7451962/querynode_1.log)

**Comment by xige-16** [comment]
The segment has a rapid increase in memory during load. The debug found that when the memory increases, the most is about 8 times the segment size. The memory increase is at the place where the memory is loaded from minio and codec deserialize

**Comment by xige-16** [comment]
> The segment has a rapid increase in memory during load. The debug found that when the memory increases, the most is about 8 times the segment size. The memory increase is at the place where the memory is loaded from minio and codec deserialize

Trouble you to check this problem @godchen0212

**Comment by binbinlv** [comment]
@xige-16 @godchen0212  any updates?

**Comment by binbinlv** [comment]
@ThreadDao please help to verify whether this is fixed by the upper pr?

**Comment by ThreadDao** [comment]
1. deploy milvus cluster `master-20211123-73f18c5` and limit querynode memory 2Gi
2. load collection (dim=512, num_entities=399360 about 780Mi)
3. search
4. release
5. re-load failed
```
[2021-11-23 04:13:43,651 - INFO - ci_test]: ################################################################################ (conftest.py:162)
[2021-11-23 04:13:43,651 - INFO - ci_test]: [initialize_milvus] Log cleaned up, start testing... (conftest.py:163)
[2021-11-23 04:13:44,673 - DEBUG - ci_test]: {
  auto_id: False
  description: 
  fields: [{
    name: int64
    description: 
    type: 5
    is_primary: True
    auto_id: False
  }, {
    name: float
    description: 
    type: 10
  }, {
    name: float_vector
    description: 
    type: 101
    params: {'dim': 512}
  }]
}
 (test_chaos_memory_stress.py:63)
[2021-11-23 04:13:44,673 - DEBUG - ci_test]: 2 (test_chaos_memory_stress.py:64)
[2021-11-23 04:13:48,047 - ERROR - ci_test]: Traceback (most recent call last):
  File "/root/milvus/tests/python_client/utils/api_request.py", line 18, in inner_wrapper
    res = func(*args, **kwargs)
  File "/root/milvus/tests/python_client/utils/api_request.py", line 45, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/orm/collection.py", line 449, in load
    conn.load_collection(self._name, timeout=timeout, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/client/stub.py", line 58, in handler
    raise e
  File "/usr/local/lib/python3.6/site-packages/pymilvus/client/stub.py", line 42, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/client/stub.py", line 322, in load_collection
    return handler.load_collection("", collection_name=collection_name, timeout=timeout, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 75, in handler
    raise e
  File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 67, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pymilvus/client/grpc_handler.py", line 820, in load_collection
    raise BaseException(response.error_code, response.reason)
pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=call query coordinator LoadCollection: rpc error: code = Unknown desc = , no queryNode to allocate)>
 (api_request.py:26)
[2021-11-23 04:13:48,047 - ERROR - ci_test]: (api_response) : <BaseException: (code=1, message=call query coordinator LoadCollection: rpc error: code = Unknown desc = , no queryNode to allocate)> (api_request.py:27)
```
[querycoord.log](https://github.com/milvus-io/milvus/files/7586174/querycoord.log)
[querynode.log](https://github.com/milvus-io/milvus/files/7586176/querynode.log)

**Comment by ThreadDao** [comment]
Fixed on `master-20211123-73f18c5`

**Comment by godchen0212** [comment]
A small optimization for this problem, when the file deserialize fails, close the binlog and event readers to prevent memory leaks.
